From 28595423ec38e88b97533cec6ec130a45c52b2fb Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Thu, 16 Jan 2014 13:47:04 +0800
Subject: [PATCH 436/505] ENGR00294868 Report sequence parameters change

Report sequence parameters change, which was enabled by API in
VPU lib v5.4.22

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/dec.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 00b38cf..91e6f36 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2013 Freescale Semiconductor, Inc.
+ * Copyright 2004-2014 Freescale Semiconductor, Inc.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -1224,8 +1224,9 @@ decoder_start(struct decode *dec)
 			outinfo.indexFrameDisplay = rotid;
 		}
 
-		dprintf(3, "frame_id %d, decidx %d, disidx %d, rotid %d\n", (int)frame_id,
-				outinfo.indexFrameDecoded, outinfo.indexFrameDisplay, rotid);
+		dprintf(3, "frame_id %d, decidx %d, disidx %d, rotid %d, decodingSuccess 0x%x\n",
+				(int)frame_id, outinfo.indexFrameDecoded, outinfo.indexFrameDisplay,
+				rotid, outinfo.decodingSuccess);
 		if (ret != RETCODE_SUCCESS) {
 			err_msg("vpu_DecGetOutputInfo failed Err code is %d\n"
 				"\tframe_id = %d\n", ret, (int)frame_id);
@@ -1267,6 +1268,9 @@ decoder_start(struct decode *dec)
 				continue;
 		}
 
+		if (cpu_is_mx6x() && (outinfo.decodingSuccess & 0x100000))
+			warn_msg("sequence parameters have been changed\n");
+
 		if (outinfo.notSufficientPsBuffer) {
 			err_msg("PS Buffer overflow\n");
 			return -1;
-- 
1.8.0

